.github/workflows/build-and-push-images.yaml

```diff
@@ -59,7 +60,63 @@ jobs:
         if: steps.publish.outcome == 'skipped'
         uses: ./.github/workflows/template-publish-image
         with:
-          image: docker.io/kubeflow/${{ matrix.component-name }}
+          image: ghcr.io/kubeflow/trainer/${{ matrix.component-name }}
+          dockerfile: ${{ matrix.dockerfile }}
+          platforms: ${{ matrix.platforms }}
+          context: ${{ matrix.context }}
+          push: false
+
+  build-and-publish-to-dockerhub:
+    name: Build and Publish Images to DockerHub
+    runs-on:
+      labels: ubuntu-latest-16-cores
+
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - component-name: trainer-controller-manager
+            dockerfile: cmd/trainer-controller-manager/Dockerfile
+            platforms: linux/amd64,linux/arm64,linux/ppc64le
+          - component-name: model-initializer
+            dockerfile: cmd/initializer/model/Dockerfile
+            platforms: linux/amd64,linux/arm64
+          - component-name: dataset-initializer
+            dockerfile: cmd/initializer/dataset/Dockerfile
+            platforms: linux/amd64,linux/arm64
+          - component-name: torchtune-trainer
+            dockerfile: cmd/trainers/torchtune/Dockerfile
+            platforms: linux/amd64,linux/arm64
+
+    steps:
+      - name: Docker Hub Login
+        if: >-
+          github.repository == 'kubeflow/trainer' &&
+          (github.ref == 'refs/heads/master' || startsWith(github.ref, 'refs/heads/release-') || startsWith(github.ref, 'refs/tags/v'))
+        uses: docker/login-action@v3
+        with:
+          registry: docker.io
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
```

Comment:
This needs to be fixed @mahdikhashan:
DOCKERHUB_USERNAME

Reply:
I'll open a hotfix.

Reply:
I mentioned fixess in follow-up PR as well